manifests/config.pp

```diff
@@ -87,4 +87,15 @@
       content => template($dns::sysconfig_template),
     }
   }
+
+  # the RedHat SCL packages do not consist of the default rfc1912 zones file
+  # therefore the file is created out of this module
+  if $facts['os']['family'] in ['RedHat'] and $dns::globals::scl {
+    file { "${dns::params::localzonepath}":
```

Comment:
No need for string interpolation. It's also better to refer to the main class rather than the private `params` class. This means it actually respects the parameter. Then you should make sure you check it's not `unmanaged`.
```suggestion
    file { $dns::localzonepath:
```